tests/execution-spec-tests/failing.llvm

```diff
@@ -1639,57 +1639,33 @@ blockchain_tests/cancun/eip6780_selfdestruct/dynamic_create2_selfdestruct_collis
+blockchain_tests/prague/eip2537_bls_12_381_precompiles/bls12_g1add/valid.json,tests/prague/eip2537_bls_12_381_precompiles/test_bls12_g1add.py::test_valid[fork_Prague-blockchain_test_from_state_test-bls_g1add_g1+p1-]
+blockchain_tests/prague/eip2537_bls_12_381_precompiles/bls12_g1add/valid.json,tests/prague/eip2537_bls_12_381_precompiles/test_bls12_g1add.py::test_valid[fork_Prague-blockchain_test_from_state_test-bls_g1add_p1+g1-]
+blockchain_tests/prague/eip2537_bls_12_381_precompiles/bls12_g2add/invalid.json,tests/prague/eip2537_bls_12_381_precompiles/test_bls12_g2add.py::test_invalid[fork_Prague-blockchain_test_from_state_test--bls_g2add_invalid_field_element-]
+blockchain_tests/prague/eip2537_bls_12_381_precompiles/bls12_g2add/invalid.json,tests/prague/eip2537_bls_12_381_precompiles/test_bls12_g2add.py::test_invalid[fork_Prague-blockchain_test_from_state_test--bls_g2add_long_input-]
+blockchain_tests/prague/eip2537_bls_12_381_precompiles/bls12_g2add/invalid.json,tests/prague/eip2537_bls_12_381_precompiles/test_bls12_g2add.py::test_invalid[fork_Prague-blockchain_test_from_state_test--bls_g2add_point_not_on_curve-]
+blockchain_tests/prague/eip2537_bls_12_381_precompiles/bls12_g2add/invalid.json,tests/prague/eip2537_bls_12_381_precompiles/test_bls12_g2add.py::test_invalid[fork_Prague-blockchain_test_from_state_test--bls_g2add_short_input-]
+blockchain_tests/prague/eip2537_bls_12_381_precompiles/bls12_g2add/invalid.json,tests/prague/eip2537_bls_12_381_precompiles/test_bls12_g2add.py::test_invalid[fork_Prague-blockchain_test_from_state_test--bls_g2add_violate_top_bytes-]
+blockchain_tests/prague/eip2537_bls_12_381_precompiles/bls12_g2add/invalid.json,tests/prague/eip2537_bls_12_381_precompiles/test_bls12_g2add.py::test_invalid[fork_Prague-blockchain_test_from_state_test--invalid_point_a_5-]
+blockchain_tests/prague/eip2537_bls_12_381_precompiles/bls12_g2add/invalid.json,tests/prague/eip2537_bls_12_381_precompiles/test_bls12_g2add.py::test_invalid[fork_Prague-blockchain_test_from_state_test--invalid_point_b_5-]
+blockchain_tests/prague/eip2537_bls_12_381_precompiles/bls12_g2msm/invalid.json,tests/prague/eip2537_bls_12_381_precompiles/test_bls12_g2msm.py::test_invalid[fork_Prague-blockchain_test_from_state_test--bls_g2_truncated_input-]
+blockchain_tests/prague/eip2537_bls_12_381_precompiles/bls12_g2mul/invalid.json,tests/prague/eip2537_bls_12_381_precompiles/test_bls12_g2mul.py::test_invalid[fork_Prague-blockchain_test_from_state_test--bls_g1_add_input_invalid_length-]
+blockchain_tests/prague/eip2537_bls_12_381_precompiles/bls12_g2mul/invalid.json,tests/prague/eip2537_bls_12_381_precompiles/test_bls12_g2mul.py::test_invalid[fork_Prague-blockchain_test_from_state_test--bls_g2_truncated_input-]
+blockchain_tests/prague/eip2537_bls_12_381_precompiles/bls12_g2mul/invalid.json,tests/prague/eip2537_bls_12_381_precompiles/test_bls12_g2mul.py::test_invalid[fork_Prague-blockchain_test_from_state_test--bls_g2mul_short_input-]

```

Comment:
I'm curious to understand why these tests are still failing. Is there some block validation we are missing here?

Reply:
I'm not sure right now. Execution fails on matching the `<gasUsed>` cell, so it might be because [EIP-7623](https://eips.ethereum.org/EIPS/eip-7623) is not implemented in this branch. I tried merging https://github.com/runtimeverification/evm-semantics/pull/2756 locally, but these proofs still fail.

Reply:
From my experience, this mismatch on call gas usually means that we're executing something we shouldn't or not executing something we should. Could this be the case here?

Reply:
Thanks for your suggestion.

Indeed, the remaining tests in `invalid.json` you see above fail because eip-7623 is not implemented in this branch. ([see this draft PR with both eips](https://github.com/runtimeverification/evm-semantics/pull/2761/files#diff-67864c9d56a0a12e77fe295edac0122a4e913d5edbaf101b3236055d4de40db5L1618-L1668)). The remaining `valid.json` tests are failing because I tried using `#asWord` instead of `Bytes2Int`. I guess that the use of `chop` is messing up the calculation. These are also addressed in the PR linked earlier. I will push a commit to revert the use of `#asWord`.

The remaining `precompile_before_fork.json` fails because we do not account for the block number when deciding which schedule is active.

Reply:
Perfect! Thanks for the clarification!